docs/DeveloperGuide.md

```diff
@@ -25,14 +29,213 @@
 
+# Implementation
+
+[Proposed] Appointment Feature
+
+The proposed appointment mechanism is facilitated by CommandHandler. It creates an appointment slot via the Appointment 
+class and recorded within the AppointmentRecord class. The appointment records will then be saved into the FileHandler
+
+- `CommandHandler.appointment(String input, AppointmentRecord appointmentRecord)` - check the input to check if valid
+- `FileHandler.initFile(AppointmentRecord appointmentRecord)` - loads any saved appointment records
+- `appointmentRecord.checkAvailability(LocalDate date, LocalTime time)` - check for availability at given time and date
+- `AppointmentRecord.addAppointment(Appointment appointment)` - add appointment to the appointment record
+- `Appointment(String name, String nric, String date, String time)` - creates appointment with the given name, nric,
+- date and time
+- `FileHandler.autosave(AppointmentRecord appointmentRecord)` - save the appointment records
+  
+
+Doctor starts up the application and executes `appointment n/John Doe ic/S123A date/18-11-2024 time/18:00`
+
+Object Diagram portray before execution of application feature:
+![img.png](BeforeExecutionOD.png)
+
+Object Diagram after execution of appointment feature:
+![img.png](AfterExecutionOD.png)
+
+The Sequence Diagram for the execution of the appointment feature:
+![img.png](NewAppointmentSD.png)
+
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+1. Should work on any mainstream OS (Windows, Linux, MacOS) as long as it has Java 17 or above installed.
+2. Should be able to hold up for long term use of a single doctor without cleaning of cache.
+3. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+4. Should be able to respond to any commands within 1 seconds under normal load
+5. Automated save of data should happen after every modification of data
 
 ## Glossary
 
-* *glossary item* - Definition
+- Mainstream OS: Windows, Linux, Unix, MacOS
+- NRIC: Identification Number
+
+## Appendix A : Instructions for Manual Testing
+
+Given below are instructions to test the app manually.
+
+<div style="background-color: #E7F3FE; padding: 12px; border-radius: 4px; border-left: 4px solid #2196F3; color: #1A1A1A; font-weight: 500;">
+⚠️ <b>Note :</b> These instructions only provide a starting point for testers to work on; testers are expected to do more <i>exploratory</i> testing.
+</div>
+
+### Command Summary Reference :
+
+| Action | Format | Example |
+|---|---|---|
+| Help | `help` | `help` |
+| Add | `add n/NAME ic/NRIC [p/PHONE_NUMBER] [d/DIAGNOSIS] [m/MEDICATION] [ha/HOME_ADDRESS] [dob/DATE_OF_BIRTH] [v/VISIT_DATE_TIME] [al/ALLERGY] [s/SEX] [mh/MEDICALHISTORY]` | `add n/James Ho ic/S9534567A p/91234567 d/Asthma m/Albuterol ha/NUS-PGPR dob/01011990 v/21-10-2024 15:48 al/Pollen s/Female mh/Diabetes` |
```

Comment:
For the appointment command syntax, should the date format dd-mm-yyyy and time format HH:mm be clarified further in the command summary?